bidict/__init__.py

```diff
@@ -50,49 +50,43 @@
 """
 
 # Use private aliases to not re-export these publicly (for Sphinx automodule with imported-members).
-from functools import partial as _partial
-from types import ModuleType as _ModuleType
-from sys import modules as _modules
-from warnings import warn as _warn
+from sys import version_info as _version_info
 
-from . import compat as _c
 
-
-if _c.PY2:
-    raise ImportError('Python 3 is required.')
-
-_warn2 = _partial(_warn, stacklevel=2)  # pylint: disable=invalid-name
-
-if (_c.PYMAJOR, _c.PYMINOR) < (3, 5):  # pragma: no cover
-    _warn2('This version of bidict is untested on Python < 3.5 and may not work.')
+if _version_info < (3, 6):  # pragma: no cover
+    raise ImportError('Python 3.6+ is required.')
 
 # The rest of this file only collects functionality implemented in the rest of the
 # source for the purposes of exporting it under the `bidict` module namespace.
-# pylint: disable=wrong-import-position
 # flake8: noqa: F401 (imported but unused)
+from ._typing import KT, VT, DT, OKT, OVT, ODT, _NONE, IterItems, MapOrIterItems
 from ._abc import BidirectionalMapping, MutableBidirectionalMapping
-from ._base import BidictBase
+from ._base import BidictBase, BT
 from ._mut import MutableBidict
 from ._bidict import bidict
 from ._frozenbidict import frozenbidict
 from ._frozenordered import FrozenOrderedBidict
 from ._named import namedbidict
 from ._orderedbase import OrderedBidictBase
 from ._orderedbidict import OrderedBidict
-from ._dup import (
-    ON_DUP_DEFAULT, ON_DUP_RAISE, ON_DUP_DROP_OLD,
-    RAISE, DROP_OLD, DROP_NEW, OnDup, OnDupAction,
-)
-from ._exc import (
-    BidictException,
-    DuplicationError, KeyDuplicationError, ValueDuplicationError, KeyAndValueDuplicationError,
-)
+from ._dup import ON_DUP_DEFAULT, ON_DUP_RAISE, ON_DUP_DROP_OLD, RAISE, DROP_OLD, DROP_NEW, OnDup, OnDupAction
+from ._exc import BidictException, DuplicationError, KeyDuplicationError, ValueDuplicationError, KeyAndValueDuplicationError
 from ._util import inverted
 from .metadata import (
     __author__, __maintainer__, __copyright__, __email__, __credits__, __url__,
     __license__, __status__, __description__, __keywords__, __version__, __version_info__,
 )
 
+# Set __module__ of re-exported classes to 'bidict' for nicer class repr strings:
```

Comment:
This feels wrong, can you provide an example of why we need this?

Reply:
Without this the private module gets exposed to the user:
```python
>>> from bidict import bidict
>>> bidict
<class 'bidict._bidict.bidict'>
```
Users should not be exposed to this implementation detail. The repr users see should match the module users are supposed to use. This change accomplishes that:
```python
>>> from bidict import bidict
>>> bidict
<class 'bidict.bidict'>
```